packages/components/src/border-control/border-control-dropdown/component.tsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { closeSmall } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import BorderControlStylePicker from '../border-control-style-picker';
+import Button from '../../button';
+import ColorIndicator from '../../color-indicator';
+import ColorPalette from '../../color-palette';
+import Dropdown from '../../dropdown';
+import { HStack } from '../../h-stack';
+import { VStack } from '../../v-stack';
+import { contextConnect, WordPressComponentProps } from '../../ui/context';
+import { useBorderControlDropdown } from './hook';
+import { StyledLabel } from '../../base-control/styles/base-control-styles';
+
+import type { DropdownProps, PopoverProps } from '../types';
+const noop = () => undefined;
+
+const BorderControlDropdown = (
+	props: WordPressComponentProps< DropdownProps, 'div' >,
+	forwardedRef: React.ForwardedRef< any >
+) => {
+	const {
+		__experimentalHasMultipleOrigins,
+		__experimentalIsRenderedInSidebar,
+		border,
+		colors,
+		disableCustomColors,
+		enableAlpha,
+		indicatorClassName,
+		indicatorWrapperClassName,
+		onReset,
+		onColorChange,
+		onStyleChange,
+		popoverClassName,
+		popoverContentClassName,
+		popoverControlsClassName,
+		resetButtonClassName,
+		enableStyle = true,
+		...otherProps
+	} = useBorderControlDropdown( props );
+
+	const { color, style } = border || {};
+
+	const renderToggle = ( { onToggle = noop } ) => (
+		<Button
+			onClick={ onToggle }
+			variant="tertiary"
+			aria-label={ __( 'Open border options' ) }
+		>
+			<span className={ indicatorWrapperClassName }>
+				<ColorIndicator
+					className={ indicatorClassName }
+					colorValue={ color }
+				/>
+			</span>
+		</Button>
+	);
+
+	const renderContent = ( { onClose }: PopoverProps ) => (
+		<>
+			<VStack className={ popoverControlsClassName } spacing={ 6 }>
+				<HStack>
+					<StyledLabel>{ __( 'Border color' ) }</StyledLabel>
+					<Button
+						isSmall
+						label={ __( 'Close border color' ) }
+						icon={ closeSmall }
+						onClick={ onClose }
+					/>
+				</HStack>
+				<ColorPalette
```

Comment:
I noted this as a known issue in the PR description and was planning to address it in a follow up to unblock some other work that depend upon this.

> Tooltips within the popover both for the close button and color palette swatches impact layout. Plan is to address in follow-up.

I'll take a run a fixing this later this evening or tomorrow.

Reply:
Ah, missed that, sorry. Not a blocker then.

Reply:
Just popping in to say this is an issue not only with this component, but with any tooltip inside a Dropdown. I could reproduce it by adding tooltips to one of the Dropdown stories:
<img width="307" alt="" src="https://user-images.githubusercontent.com/8096000/159634250-e9cecd9a-c115-4910-aa15-c75c46062542.png">

It can also be seen in the ColorPalette story, no changes needed:
<img width="340" alt="" src="https://user-images.githubusercontent.com/8096000/159634402-9ce390d9-474b-450f-8e82-69ae25f58fce.png">

It only seems to happen within Storybook though; where these components are used in the editor the tooltips behave as expected.

Reply:
Thanks for taking the time to investigate this as well @tellthemachines 👍 

I've finally found what I was missing here to make the tooltips render correctly. I needed to wrap the storybook example in a slot fill provider and include the `Popover.Slot` as well.

This was done in https://github.com/WordPress/gutenberg/pull/37769/commits/535cd0dc5851ccd6fcd99e9d4023f9bd17195788.

https://user-images.githubusercontent.com/60436221/159641208-f6709997-93cc-4968-b490-06d4704bd34a.mp4